src/geode/geosciences_io/mesh/geotiff_input.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2019 - 2025 Geode-solutions
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/geosciences_io/mesh/internal/geotiff_input.hpp>
+
+#include <ogrsf_frmts.h>
+
+#include <geode/basic/attribute_manager.hpp>
+#include <geode/basic/file.hpp>
+#include <geode/basic/filename.hpp>
+#include <geode/basic/logger.hpp>
+
+#include <geode/mesh/core/light_regular_grid.hpp>
+
+#include <geode/geometry/point.hpp>
+#include <geode/geometry/vector.hpp>
+
+#include <geode/image/core/raster_image.hpp>
+#include <geode/image/core/rgb_color.hpp>
+#include <geode/image/io/raster_image_output.hpp>
+
+#include <geode/io/image/internal/tiff_input.hpp>
+
+namespace
+{
+    class GEOTIFFInputImpl
+    {
+    public:
+        GEOTIFFInputImpl( std::string_view filename )
+            : gdal_data_{ GDALDataset::Open(
+                  geode::to_string( filename ).c_str(), GDAL_OF_VECTOR ) }
+        {
+            OPENGEODE_EXCEPTION(
+                gdal_data_, "[GEOTIFFInput] Failed to open file ", filename );
+            get_geotransform();
+        }
+
+        const geode::Point2D& get_origin() const
+        {
+            return origin_;
+        }
+
+        std::array< geode::Vector2D, 2 > get_directions() const
+        {
+            return { x_direction_, y_direction_ };
+        }
+
+        bool get_geotransform()
+        {
+            double geoTransform[6];
+            if( gdal_data_->GetGeoTransform( geoTransform ) == CE_Failure )
+            {
+                return false;
```

Comment:
can it failed silently or is this a reason to exception ? Can we load the data without this ?

Reply:
yes because by default the origin is (0,0) and vectors are (1,0) and (0,1).